simulation/src/main/java/com/palantir/dialogue/core/HistogramChannel.java

```diff
@@ -46,7 +46,13 @@ public Histogram getHistogram() {
     public ListenableFuture<Response> execute(Endpoint endpoint, Request request) {
         long start = simulation.clock().read();
         ListenableFuture<Response> future = channel.execute(endpoint, request);
-        future.addListener(() -> histogram.update(simulation.clock().read() - start), MoreExecutors.directExecutor());
+        future.addListener(
+                () -> {
+                    long micros =
+                            TimeUnit.MICROSECONDS.convert(simulation.clock().read() - start, TimeUnit.NANOSECONDS);
+                    histogram.update(micros);
```

Comment:
can we use a timer instead?

Reply:
Have updated to use the vanilla InstrumentedChannel!

Reply:
Thanks!